internal/queue/pending_txs.go

```diff
@@ -57,27 +64,25 @@ func (p *PendingTxs) DoneWithReceipt(txIDs []string, blockHeader *types.BlockHea
 func (p *PendingTxs) ReleaseWithError(txIDs []string, err error) {
 	p.logger.Debugf("Release with error: %s; txIDs: %v", err, txIDs)
 
-	p.Lock()
-	defer p.Unlock()
-
 	for _, txID := range txIDs {
-		p.txs[txID].error(err)
-
-		delete(p.txs, txID)
+		promise, loaded := p.loadAndDelete(txID)
+		if !loaded {
+			continue
+		}
+		promise.error(err)
 	}
 }
 
 func (p *PendingTxs) Has(txID string) bool {
-	p.RLock()
-	defer p.RUnlock()
-
-	_, ok := p.txs[txID]
+	_, ok := p.txs.Load(txID)
 	return ok
 }
 
 func (p *PendingTxs) Empty() bool {
-	p.RLock()
-	defer p.RUnlock()
-
-	return len(p.txs) == 0
+	empty := true
+	p.txs.Range(func(key, value interface{}) bool {
+		empty = false
+		return false
+	})
+	return empty
```

Comment:
This has semantics different than the previous Empty(), as Range does not respect any snapshot. It is probably OK to use in tests, but not in production (we only use in tests, lets keep it this way). Maybe we should put a comment on the method in line with the comment on Range.

Reply:
Outdated. Reverted back to map.